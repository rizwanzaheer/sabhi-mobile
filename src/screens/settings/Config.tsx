/**
 * Serto Mobile App
 *
 */
import React from 'react'
import { Container, Screen, ListItem, Section } from '@kancha/kancha-ui'
import Config from 'react-native-config'

export default () => {
  const configs = Object.keys(Config.getConstants())

  // tslint:disable-next-line:no-console
  console.log('Configs', configs)

  return (
    <Screen scrollEnabled={true}>
      <Container>
        <Section>
          {configs.map((key: string, i: number) => {
            const value = Config[key].toString()
            return (
              <ListItem
                key={key}
                subTitle={key}
                last={i === configs.length - 1}
              >
                {value.substring(0, 25) + (value.length > 25 ? '...' : '')}
              </ListItem>
            )
          })}
        </Section>
      </Container>
    </Screen>
  )
}
