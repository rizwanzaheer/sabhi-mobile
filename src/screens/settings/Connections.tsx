/**
 * Serto Mobile App
 *
 */

import React from 'react'
import { FlatList, Image } from 'react-native'
import { Query, QueryResult } from 'react-apollo'
import {
  Container,
  Screen,
  ListItem,
  Text,
  Avatar,
  Constants,
  useTheme,
} from '@kancha/kancha-ui'
import { NavigationStackScreenProps } from 'react-navigation-stack'
import { GET_ALL_IDENTITIES } from '../../lib/graphql/queries'

interface Result extends QueryResult {
  data: { identities: any[] }
}

interface Props extends NavigationStackScreenProps {}

const Connections: React.FC<Props> = props => {
  const theme = useTheme()

  return (
    <Screen safeArea={true}>
      <Container flex={1}>
        <Query query={GET_ALL_IDENTITIES} onError={console.log}>
          {({ data, loading, refetch, error }: Result) =>
            error ? (
              <Text>{error.message}</Text>
            ) : (
              <FlatList
                style={{ flex: 1 }}
                data={data && data.identities}
                renderItem={({ item, index }) => {
                  const imgSrc = item.profileImage
                    ? { source: { uri: item.profileImage } }
                    : {}
                  return (
                    <ListItem
                      iconLeft={
                        <Avatar
                          {...imgSrc}
                          address={item.did}
                          type={'circle'}
                          gravatarType={'retro'}
                        />
                      }
                      // onPress={() => {
                      //   props.navigation.push('Credentials', {
                      //     did: item.did,
                      //   })
                      // }}
                      last={index === data.identities.length - 1}
                    >
                      {item.shortId}
                    </ListItem>
                  )
                }}
                keyExtractor={item => item.did}
                onRefresh={refetch}
                refreshing={loading}
                ListEmptyComponent={
                  <Container padding>
                    <Text
                      type={Constants.TextTypes.H3}
                      bold
                      textColor={theme.colors.primary.text}
                    >
                      No Connections
                    </Text>
                  </Container>
                }
              />
            )
          }
        </Query>
      </Container>
    </Screen>
  )
}

export default Connections
