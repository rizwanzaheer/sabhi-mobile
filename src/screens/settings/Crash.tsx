import * as React from 'react'
import { Screen, Container, Button, Constants } from '@kancha/kancha-ui'
// import * as Sentry from '@sentry/react-native'

export default () => {
  return (
    <Screen scrollEnabled>
      <Container padding>
        <Button
          fullWidth
          type={Constants.BrandOptions.Warning}
          block={Constants.ButtonBlocks.Outlined}
          buttonText={'JS Crash'}
          onPress={() => {
            throw new Error('Sample error from developer tools')
          }}
        />
      </Container>
      <Container padding>
        <Button
          fullWidth
          type={Constants.BrandOptions.Warning}
          block={Constants.ButtonBlocks.Outlined}
          buttonText={'Native Crash'}
          onPress={() => {
            console.log('onPress');
            // Sentry.nativeCrash()
          }}
        />
      </Container>
    </Screen>
  )
}
