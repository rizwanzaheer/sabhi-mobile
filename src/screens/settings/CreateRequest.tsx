/**
 * Serto Mobile App
 */
import React, { useState, useEffect } from 'react'
import {
  Container,
  Button,
  Constants,
  Screen,
  Text,
  Icon,
  RadioBtn,
} from '@kancha/kancha-ui'
import { TextInput } from 'react-native'
import { useNavigation } from 'react-navigation-hooks'
import { useQuery } from 'react-apollo'
import { GET_VIEWER, GET_ALL_IDENTITIES } from '../../lib/graphql/queries'

import { Colors } from '../../theme'
import { TouchableHighlight } from 'react-native-gesture-handler'

interface Claim {
  claimType: string
  reason: string
  essential: boolean
}

/**
 * Temporary implementation until all requirements have been ironed out.
 * Final component will be built in @kancha
 */
export default () => {
  const navigation = useNavigation()

  const { data } = useQuery(GET_VIEWER)
  const getAllIdentitiesResp = useQuery(GET_ALL_IDENTITIES)

  const [identities, setIdentities] = useState<any[]>([])
  const [subject, updateSubject] = useState<any>({})
  const [identityBrowseOpen, toggleIdentityBrowse] = useState(false)

  const [claims, updateClaims] = useState<any>([])
  const [claimType, updateClaimType] = useState()
  const [claimTypeRequired, updateClaimTypeRequired] = useState(false)
  const [claimReason, updateClaimReason] = useState()
  const [claimTypeIssuers, updateClaimTypeIssuers] = useState([])

  const addClaimField = (type: string, reason: string, essential: boolean) => {
    const field = {
      claimType: type,
      reason,
      essential,
    }
    const updatedClaims = claims.concat([field])
    updateClaims(updatedClaims)
    updateClaimType('')
    updateClaimReason('')
    updateClaimTypeRequired(false)
  }

  const removeClaimField = (index: number) => {
    const updatedClaims = claims.filter((item: any, i: number) => i !== index)
    updateClaims(updatedClaims)
  }

  const signClaim = () => {
    navigation.navigate('SendRequest', {
      request: {
        claims,
        tag: 'tag-' + Date.now().toString(),
        issuer: data.viewer.did,
        subject: subject && subject.did,
      },
    })
  }

  useEffect(() => {
    if (getAllIdentitiesResp.data && getAllIdentitiesResp.data.identities) {
      setIdentities(getAllIdentitiesResp.data.identities)
    }
  }, [data])

  return (
    <Screen scrollEnabled={true} background={'primary'}>
      <Container padding>
        <Container marginBottom>
          <Text type={Constants.TextTypes.H3} bold>
            Create Request
          </Text>
          <Container marginTop>
            <Container>
              <Text type={Constants.TextTypes.SubTitle}>
                Request information from: (Optional)
              </Text>
            </Container>

            <Container background={'secondary'} padding marginTop br={5}>
              <TextInput
                placeholder={'Enter did'}
                autoCorrect={false}
                autoCapitalize={'none'}
                autoCompleteType={'off'}
                placeholderTextColor={Colors.DARK_GREY}
                style={{ color: Colors.BRAND }}
                onFocus={() => toggleIdentityBrowse(true)}
                onBlur={() => toggleIdentityBrowse(false)}
                onChangeText={value =>
                  updateSubject({ did: value, shortId: 'Unknown' })
                }
                value={subject.did}
              ></TextInput>
              {identityBrowseOpen && (
                <Container marginTop dividerBottom>
                  {identities.map((id: any) => {
                    return (
                      <TouchableHighlight
                        underlayColor={Colors.LIGHTEST_GREY}
                        key={id.did}
                        onPress={() => {
                          updateSubject(id)
                          toggleIdentityBrowse(false)
                        }}
                      >
                        <Container dividerTop paddingTop paddingBottom>
                          <Text bold>{id.shortId}</Text>
                          <Text textStyle={{ fontSize: 12 }}>{id.did}</Text>
                        </Container>
                      </TouchableHighlight>
                    )
                  })}
                </Container>
              )}
            </Container>
          </Container>
          <Container marginTop>
            <Text type={Constants.TextTypes.SectionHeader}>
              <Text type={Constants.TextTypes.SubTitle}>
                This request will be sent from:{' '}
                <Text bold>{data && data.viewer && data.viewer.shortId}</Text>
              </Text>
            </Text>
          </Container>
        </Container>
        <Container>
          {claims.map((c: Claim, i: number) => {
            return (
              <Container
                key={i}
                backgroundColor={Colors.CHARCOAL}
                br={8}
                padding
                marginBottom
              >
                <Text textColor={Colors.WHITE} textStyle={{ fontSize: 12 }}>
                  {c.claimType}
                </Text>
                <Text textColor={Colors.WHITE} textStyle={{ fontSize: 12 }}>
                  {c.reason}
                </Text>
                <Text textColor={Colors.WHITE} warn={c.essential}>
                  {c.essential ? 'Required' : 'Not Required'}
                </Text>
                <Container
                  viewStyle={{ position: 'absolute', top: 20, right: 10 }}
                >
                  <Button
                    icon={
                      <Icon
                        size={25}
                        icon={{ name: 'ios-close', iconFamily: 'Ionicons' }}
                      />
                    }
                    iconButton
                    onPress={() => removeClaimField(i)}
                  ></Button>
                </Container>
              </Container>
            )
          })}
        </Container>
        <Container backgroundColor={Colors.LIGHTEST_GREY} br={8} padding>
          <Container backgroundColor={Colors.WHITE} padding br={8}>
            <TextInput
              autoCorrect={false}
              autoCapitalize={'none'}
              autoCompleteType={'off'}
              placeholderTextColor={Colors.LIGHT_GREY}
              placeholder={'Enter claim type'}
              value={claimType}
              style={{ fontSize: 12 }}
              onChangeText={updateClaimType}
            ></TextInput>
          </Container>
          <Container
            backgroundColor={Colors.WHITE}
            padding
            br={8}
            marginTop={10}
          >
            <TextInput
              autoCorrect={false}
              autoCapitalize={'none'}
              autoCompleteType={'off'}
              placeholderTextColor={Colors.LIGHT_GREY}
              placeholder={'Enter reason for requesting'}
              value={claimReason}
              style={{ fontSize: 12 }}
              onChangeText={updateClaimReason}
            ></TextInput>
          </Container>
          <Container
            paddingTop
            flexDirection={'row'}
            alignItems={'center'}
            justifyContent={'space-between'}
          >
            <Container flex={1}>
              <RadioBtn
                onPress={() => updateClaimTypeRequired(!claimTypeRequired)}
                selected={claimTypeRequired}
              >
                Required
              </RadioBtn>
            </Container>
            <Container justifyContent={'center'}>
              <Button
                disabled={!claimType}
                block={Constants.ButtonBlocks.Filled}
                type={Constants.BrandOptions.Primary}
                buttonText={'+ Add'}
                small
                onPress={() =>
                  addClaimField(claimType, claimReason, claimTypeRequired)
                }
              ></Button>
            </Container>
          </Container>
        </Container>

        <Container marginTop>
          <Container>
            <Button
              buttonText={subject.did ? 'Send Request' : 'Generate QRCode'}
              fullWidth
              type={'confirm'}
              block={'filled'}
              onPress={() => signClaim()}
              disabled={claims.length === 0}
              shadowOpacity={0.2}
            ></Button>
          </Container>
        </Container>
      </Container>
    </Screen>
  )
}
