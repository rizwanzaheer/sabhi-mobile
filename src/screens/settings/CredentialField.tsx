import React, { useState, useEffect } from 'react'
import {
  Screen,
  Container,
  Text,
  Constants,
  Button,
  Device,
} from '@kancha/kancha-ui'
import { ActivityIndicator } from 'react-native'
import { useNavigationParam } from 'react-navigation-hooks'
import { useQuery } from '@apollo/react-hooks'
import { GET_MESSAGE } from '../../lib/graphql/queries'
import QRCode from 'react-native-qrcode-svg'

interface MessageProcess {}

const MessageProcess: React.FC<MessageProcess> = () => {
  const hash = useNavigationParam('hash')
  const { data, loading } = useQuery(GET_MESSAGE, {
    variables: {
      hash,
    },
  })

  return (
    <Screen safeAreaBottom>
      <Container flex={1} alignItems={'center'} justifyContent={'center'}>
        {loading && <ActivityIndicator size={'large'} />}
        {!loading && data && data.message && data.message.jwt && (
          <QRCode value={data.message.jwt} size={Device.width - 100} />
        )}
      </Container>
    </Screen>
  )
}

export default MessageProcess
