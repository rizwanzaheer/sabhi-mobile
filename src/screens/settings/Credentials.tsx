/**
 * Serto Mobile App
 *
 */
import React from 'react'
import { useTranslation } from 'react-i18next'
import { FlatList } from 'react-native'
import { useQuery } from '@apollo/react-hooks'
import {
  Container,
  Screen,
  ListItem,
  Text,
  Avatar,
  Constants,
} from '@kancha/kancha-ui'
import { Colors } from '../../theme'
import { withNavigation } from 'react-navigation'
import { NavigationStackScreenProps } from 'react-navigation-stack'
import { GET_CREDENTIALS_FOR_IDENTITY } from '../../lib/graphql/queries'

interface Props extends NavigationStackScreenProps {}

export const Credentials: React.FC<Props> = props => {
  const { navigation } = props
  const did = navigation.getParam('did')
  const { data, loading, error, refetch } = useQuery(
    GET_CREDENTIALS_FOR_IDENTITY,
    {
      variables: {
        did,
      },
    },
  )

  return (
    <Container />
    // <Screen safeArea={true}>
    //   <Container flex={1}>
    //     {error ? (
    //       <Text>Error</Text>
    //     ) : (
    //       <FlatList
    //         style={{ backgroundColor: Colors.LIGHTEST_GREY, flex: 1 }}
    //         data={data && data.identity && data.identity.receivedCredentials}
    //         renderItem={({ item, index }: { item: any; index: number }) => {
    //           const imgSrc = item.issuer.profileImage
    //             ? { source: { uri: item.issuer.profileImage } }
    //             : {}
    //           const claims = item.claims.map((field: any) => field)
    //           return claims.map((claim: any, index: number) => {
    //             return (
    //               <ListItem
    //                 key={item.hash}
    //                 subTitle={claims.type}
    //                 accessoryRight={
    //                   item.issuer.did === item.subject.did
    //                     ? 'Self signed'
    //                     : item.issuer.shortId
    //                 }
    //                 iconLeft={
    //                   <Avatar
    //                     {...imgSrc}
    //                     address={item.issuer.did}
    //                     type={'circle'}
    //                     gravatarType={'retro'}
    //                   />
    //                 }
    //                 last={index === data.credentials.length - 1}
    //               >
    //                 {claims.value.startsWith('https://images.app') ||
    //                 claims.value.endsWith('.jpg') ||
    //                 claims.value.endsWith('.png')
    //                   ? '<image url>'
    //                   : claims.value}
    //               </ListItem>
    //             )
    //           })
    //         }}
    //         keyExtractor={item => item.hash}
    //         onRefresh={refetch}
    //         refreshing={loading}
    //         ListEmptyComponent={
    //           !loading ? (
    //             <Container padding>
    //               <Text
    //                 type={Constants.TextTypes.H3}
    //                 bold
    //                 textColor={Colors.DARK_GREY}
    //               >
    //                 No Credentials
    //               </Text>
    //             </Container>
    //           ) : (
    //             <Container />
    //           )
    //         }
    //       />
    //     )}
    //   </Container>
    // </Screen>
  )
}

export default withNavigation(Credentials)
