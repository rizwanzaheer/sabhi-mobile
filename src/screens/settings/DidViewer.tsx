import * as React from 'react'
import { Screen, Container, Text, Constants, Section } from '@kancha/kancha-ui'
import { withNavigation } from 'react-navigation'
import { NavigationStackScreenProps } from 'react-navigation-stack'

interface DidViewerProps extends NavigationStackScreenProps {}

export const DidViewer: React.FC<DidViewerProps> = props => {
  const { navigation } = props
  const did = navigation.getParam('did', 'Did does not exist anymore')

  return (
    <Screen
      scrollEnabled
      background={Constants.BrandOptions.Secondary}
      safeArea
    >
      <Container>
        <Section title={'DID'}>
          <Container padding>
            <Text>{did}</Text>
          </Container>
        </Section>
      </Container>
    </Screen>
  )
}

export default withNavigation(DidViewer)
