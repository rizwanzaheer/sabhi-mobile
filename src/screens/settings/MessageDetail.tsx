import * as React from 'react'
import {
  Container,
  Screen,
  Text,
  Constants,
  Avatar,
  Icon,
  Device,
} from '@kancha/kancha-ui'
import { NavigationScreen } from '../../navigators'
import JSONTree from 'react-native-json-tree'
import { useNavigationParam } from 'react-navigation-hooks'
import { formatDistanceToNow } from 'date-fns'
import { Colors } from '../../theme'
import QRCode from 'react-native-qrcode-svg'

const Component: React.FC<NavigationScreen> = () => {
  const message = useNavigationParam('message')

  const issProfileSource = message.from.profileImage
    ? { source: { uri: message.from.profileImage } }
    : {}
  const subProfileSource =
    message.to && message.to.profileImage
      ? { source: { uri: message.to.profileImage } }
      : {}

  return (
    <Screen scrollEnabled={true} background={'primary'}>
      <Container padding marginTop>
        <Text type={Constants.TextTypes.H2} bold>
          Message Detail
        </Text>
        <Container marginTop={5} marginBottom>
          <Text type={Constants.TextTypes.ActivitySubTitle}>
            {(message.saveDate && formatDistanceToNow(message.saveDate)) +
              ' ago' || 'Some time ago'}
            {' • '}
            Message type: {message.type}
            {' • '}
            Thread ID: {message.threadId}
          </Text>
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Identities
            </Text>
          </Container>
          {/* <Container
            background={'secondary'}
            paddingTop={32}
            br={10}
            paddingBottom={32}
            flexDirection={'row'}
            flex={1}
            justifyContent={'center'}
            alignItems={'center'}
            paddingLeft={32}
            paddingRight={32}
            marginBottom
          >
            <Container alignItems={'center'}>
              <Avatar
                {...issProfileSource}
                type={'circle'}
                gravatarType={'retro'}
                address={message.from.did}
                size={50}
              />
              <Container paddingTop={8}>
                <Text type={Constants.TextTypes.ActivityTitle}>
                  {message.from.shortId}
                </Text>
              </Container>
            </Container>
            {message.to && (
              <Container>
                <Container
                  flexDirection={'row'}
                  marginLeft={25}
                  marginRight={25}
                >
                  <Icon
                    icon={{ name: 'ios-arrow-forward', iconFamily: 'Ionicons' }}
                    size={30}
                  />
                </Container>
                <Container alignItems={'center'}>
                  <Avatar
                    {...subProfileSource}
                    type={'circle'}
                    gravatarType={'retro'}
                    address={message.to.did}
                    size={50}
                  />
                  <Container paddingTop={8}>
                    <Text type={Constants.TextTypes.ActivityTitle}>
                      {message.to.shortId}
                    </Text>
                  </Container>
                </Container>
              </Container>
            )}
          </Container> */}
          <Container
            background={'secondary'}
            br={10}
            padding
            marginBottom
            flexDirection={'row'}
            alignItems={'center'}
          >
            <Avatar
              {...issProfileSource}
              type={'circle'}
              gravatarType={'retro'}
              address={message.from.did}
              size={50}
            />
            <Container paddingLeft flex={1}>
              <Text selectable>
                <Text
                  textStyle={{ flexWrap: 'wrap' }}
                  type={Constants.TextTypes.SubTitle}
                >
                  from:
                </Text>
                {message.from.did}
              </Text>
            </Container>
          </Container>
          {message.to && (
            <Container
              background={'secondary'}
              br={10}
              padding
              marginBottom
              flexDirection={'row'}
              alignItems={'center'}
            >
              <Avatar
                {...subProfileSource}
                type={'circle'}
                gravatarType={'retro'}
                address={message.to.did}
                size={50}
              />
              <Container paddingLeft flex={1}>
                <Text selectable>
                  <Text type={Constants.TextTypes.SubTitle}>to:</Text>
                  {message.to.did}
                </Text>
              </Container>
            </Container>
          )}
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Attachments (Credentials)
            </Text>
          </Container>
          {message.credentials.map((vc: any, index: number) => {
            return (
              <Container
                key={index}
                background={'secondary'}
                br={10}
                padding
                marginBottom
              >
                <JSONTree
                  hideRoot={true}
                  data={vc.claims}
                  invertTheme={false}
                  theme={{
                    scheme: 'brewer',
                    base00: Colors.LIGHTEST_GREY,
                    base01: '#2e2f30',
                    base02: '#515253',
                    base03: '#737475',
                    base04: '#959697',
                    base05: '#b7b8b9',
                    base06: '#dadbdc',
                    base07: '#fcfdfe',
                    base08: '#e31a1c',
                    base09: '#e6550d',
                    base0A: '#dca060',
                    base0B: '#31a354',
                    base0C: '#80b1d3',
                    base0D: '#3182bd',
                    base0E: '#756bb1',
                    base0F: '#b15928',
                  }}
                />
              </Container>
            )
          })}
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Meta Data
            </Text>
          </Container>
          <Container background={'secondary'} br={10} padding marginBottom>
            <JSONTree
              hideRoot={true}
              data={message.metaData}
              invertTheme={false}
              theme={{
                scheme: 'brewer',
                base00: Colors.LIGHTEST_GREY,
                base01: '#2e2f30',
                base02: '#515253',
                base03: '#737475',
                base04: '#959697',
                base05: '#b7b8b9',
                base06: '#dadbdc',
                base07: '#fcfdfe',
                base08: '#e31a1c',
                base09: '#e6550d',
                base0A: '#dca060',
                base0B: '#31a354',
                base0C: '#80b1d3',
                base0D: '#3182bd',
                base0E: '#756bb1',
                base0F: '#b15928',
              }}
            />
          </Container>
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Message ID
            </Text>
          </Container>
          <Container background={'secondary'} br={10} padding marginBottom>
            <Text selectable>{message.id}</Text>
          </Container>
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Raw Message
            </Text>
          </Container>
          <Container background={'secondary'} br={10} padding marginBottom>
            <JSONTree
              hideRoot={true}
              data={message.data}
              invertTheme={false}
              theme={{
                scheme: 'brewer',
                base00: Colors.LIGHTEST_GREY,
                base01: '#2e2f30',
                base02: '#515253',
                base03: '#737475',
                base04: '#959697',
                base05: '#b7b8b9',
                base06: '#dadbdc',
                base07: '#fcfdfe',
                base08: '#e31a1c',
                base09: '#e6550d',
                base0A: '#dca060',
                base0B: '#31a354',
                base0C: '#80b1d3',
                base0D: '#3182bd',
                base0E: '#756bb1',
                base0F: '#b15928',
              }}
            />
          </Container>
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Message JWT
            </Text>
          </Container>
          <Container background={'secondary'} br={10} padding marginBottom>
            <Text selectable>{message.raw}</Text>
          </Container>
        </Container>
        <Container>
          <Container marginBottom>
            <Text type={Constants.TextTypes.H5} bold>
              Message QRCode
            </Text>
          </Container>
          <Container alignItems={'center'} padding marginBottom>
            <QRCode value={message.raw} size={Device.width - 100} />
          </Container>
        </Container>
      </Container>
    </Screen>
  )
}

export default Component
