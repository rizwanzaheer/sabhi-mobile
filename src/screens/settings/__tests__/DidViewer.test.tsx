import 'react-native'
import React from 'react'
import { DidViewer } from '../../settings/DidViewer'
import { MockedProvider } from '@apollo/react-testing'
import { render } from 'react-native-testing-library'

jest.useFakeTimers()
jest.runAllTimers()

const navigation = {
  push: jest.fn(),
  getParam: jest.fn().mockReturnValue('0x1234'),
  goBack: jest.fn(),
}

test('renders correctly', () => {
  // jest.useFakeTimers()
  // @ts-ignore
  const screen = <DidViewer navigation={navigation} />
  const tree = render(
    <MockedProvider mocks={[]}>{screen}</MockedProvider>,
  ).toJSON()

  // jest.runAllTimers()
  expect(tree).toMatchSnapshot()
})
