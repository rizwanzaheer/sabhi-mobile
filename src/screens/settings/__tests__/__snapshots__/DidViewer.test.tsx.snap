// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />,
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    contentInsetAdjustmentBehavior="never"
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "#EFEEF3",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#EFEEF3",
            "borderRadius": undefined,
            "flex": 1,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": 20,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": 15,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#BBBBBB",
                    "flexWrap": "wrap",
                    "fontSize": 14,
                    "paddingBottom": 15,
                  }
                }
              >
                DID
              </Text>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "#CCCCCC",
                  "borderBottomWidth": 0.5,
                  "borderRadius": undefined,
                  "borderTopColor": "#CCCCCC",
                  "borderTopWidth": 0.5,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                    }
                  }
                >
                  0x1234
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  />,
]
`;
